File: cpp/DIP/step2/WeatherTracker.cpp
#include <iostream>
#include <string>
#include "Phone.cpp"
#include "Emailer.cpp"
using namespace std;

class WeatherTracker {
	WeatherTracker() {
	}

	void SetCurrentConditions(string weather_description) {
		current_conditions_ = weather_description;
		if (weather_description == "rainy") {
			string alert = phone_.GenerateWeatherAlert(weather_description);
			cout << alert << "\n";
		}
		if (weather_description == "sunny") {
			string alert = emailer_.GenerateWeatherAlert(weather_description);
			cout << alert << "\n";
		}
	}

private:
	string current_conditions_;
	Phone phone_;
	Emailer emailer_;
};

File: python/DIP/step2/weather_tracker.py
from phone import Phone
from emailer import Emailer

class WeatherTracker:
    def __init__(self) -> None:
        super().__init__()
        self.__current_conditions = ""
        self.__phone = Phone()
        self.__emailer = Emailer()

    def set_current_conditinos(self, weather_description: str) -> None:
        self.__current_conditions = weather_description
        if weather_description == 'rainy':
            alert = self.__phone.generate_weather_alert(weather_description)
            print(alert)
        if weather_description == 'sunny':
            alert = self.__emailer.generate_weather_alert(weather_description)
            print(alert)
